react/src/pages/addParticipants/DataEntryTable.tsx

```diff
@@ -101,6 +119,10 @@ export default function DataEntryTable(props: DataEntryTableProps) {
     const [files, setFiles] = useState<string[]>([]);
 
     const [showRNA, setShowRNA] = useState<boolean>(false);
+    // rows where participant associated columns are disabled
+    const [disabledRows, setDisabledRows] = useState<number[]>([]);
```

Comment:
Storing the row index of disabled rows separately will lead to issues since the number of rows changes dynamically. Duplicates of disabled rows are not disabled, and deleting a disabled row in the middle of the table will cause the next row to be disabled when it takes the deleted row's spot.

I would suggest instead to adding a disabled field to the DataEntryRow interface, since whether a row is disabled or not makes sense as a property of the row.